python/pyspark/ml/torch/distributor.py

```diff
@@ -744,7 +800,44 @@ def run(self, train_object: Union[Callable, str], *args: Any) -> Optional[Any]:
                 TorchDistributor._run_training_on_pytorch_function  # type: ignore
             )
         if self.local_mode:
-            output = self._run_local_training(framework_wrapper_fn, train_object, *args)
+            output = self._run_local_training(framework_wrapper_fn, train_object, *args, **kwargs)
         else:
-            output = self._run_distributed_training(framework_wrapper_fn, train_object, *args)
+            output = self._run_distributed_training(
+                framework_wrapper_fn, train_object, None, *args, **kwargs
+            )
         return output
+
+    def train_on_dataframe(self, train_function, spark_dataframe, *args, **kwargs):
+        if self.local_mode:
+            raise ValueError(
+                "`TorchDistributor.train_on_dataframe` requires setting `TorchDistributor.local_mode` to `False`."
+            )
+
+        return self._run_distributed_training(
+            TorchDistributor._run_training_on_pytorch_function,
+            train_function,
+            spark_dataframe,
+            *args,
+            **kwargs
+        )
+
+
+def get_spark_partition_data_loader(num_samples, batch_size, prefetch=2):
```

Comment:
why `prefetch=2`?

Reply:
This is the default value in pytorch DataLoader.